src/main/java/org/folio/circulation/infrastructure/storage/inventory/InstanceRepository.java

```diff
@@ -55,37 +54,27 @@ public InstanceRepository(Clients clients) {
       return completedFuture(succeeded(multipleRequests));
     }
 
-    final FindWithMultipleCqlIndexValues<Instance> fetcher = createInstancesFetcher();
-
-    return fetcher.findByIds(instanceIdsToFetch)
+    return  findWithMultipleCqlIndexValues(instancesClient, "instances", Instance::from)
+      .findByIds(instanceIdsToFetch)
       .thenApply(multipleInstancesResult -> multipleInstancesResult.next(
         multipleInstances -> {
-          List<Request> newRequestList = new ArrayList<>();
-          Collection<Instance> instanceCollection = multipleInstances.getRecords();
-          for (Request request : requests) {
-            Request newRequest = null;
-            boolean foundInstance = false;
-            for (Instance instance : instanceCollection) {
-              if (request.getInstanceId() != null && request.getInstanceId().equals(instance.getId())) {
-                newRequest = request.withInstance(instance);
-                foundInstance = true;
-                break;
+          List<Request> newRequestList = requests.stream()
+            .map(request -> {
+              Optional<Instance> matchedInstance = multipleInstances.getRecords().stream()
+                .filter(instance -> (request.getInstanceId() != null && request.getInstanceId().equals(instance.getId())))
+                .findFirst();
+              if (matchedInstance.isEmpty()) {
+                log.info("No instance found for request {} (instanceId {})", request.getId(), request.getInstanceId());
+                return request;
+              } else {
```

Comment:
Can this if/else be replaces with ifPresentOrElse maybe?

Reply:
ifPresentOrElse would be nice, but its return type is void, but we need to return Request type. So that, I am not sure we can use ifPresentOrElse in this case

Reply:
done